### ISSUE TITLE
ICLR '18 | On the information bottleneck theory of deep learning.

### ISSUE THREAD

**Issue by Fassial** [Fassial/read-papers#12]
https://openreview.net/pdf?id=ry_WPG-A-

**Comment by Fassial** [comment]
## 相关文献
 - Tishby N, Zaslavsky N. [Deep learning and the information bottleneck principle.](http://robotics.caltech.edu/wiki/images/8/8f/DeepLearningBottleneck.pdf)
 - Shwartz-Ziv R, Tishby N. [Opening the black box of deep neural networks via information.](https://arxiv.org/pdf/1703.00810.pdf)

**Comment by Fassial** [comment]
## 特色
用information bottleneck（来源于信息论）这一套理论来解释深度学习，包括深度学习的实质、泛化等。Tishby先提出（可能不是第一个）用信息论的这一套方法来分析深度学习，后面被Shwartz-Ziv的文章拓展。由于第二篇文章比较详细，这里重点读第二篇文章。后面2018年的ICLR文章又对于相关的结论进行了一些反驳，挺有意思的，挂出来作为参考。
为什么关注该问题？在深度学习领域，有这么一套信息论方法用来描述什么是“最优”表示；而强化学习里面的一个核心研究内容是representation learning，但是在强化学习里面大家对于什么是一个“好”的representation还没有很好的概念。我们希望能够参考这一理论给出一些回答。

**Comment by Fassial** [comment]
## 背景：互信息
给定两个随机变量X、Y，互信息反映了一个随机变量包含了另一个随机变量的多少信息，单位一般为bit。
![v2-c24c3d7dc2e9d5ddd2988e36ad2bf2e4_720w](https://user-images.githubusercontent.com/37235023/147567157-f6e180c4-232e-48d2-8b41-0e4f8eb5d692.jpg)
**性质1**：可逆变换不改变信息量
<img width="744" alt="v2-a803b67708f071a4d756413d5126dc4d_720w" src="https://user-images.githubusercontent.com/37235023/147567287-b2b274de-7df8-40c4-8fda-32823d2002ae.png">
**性质2**：数据处理不等式（data processing inequality, DPI）对于如下的数据生成Markov Chain（X->Y->Z），有
<img width="730" alt="v2-595786503179f7c6ff439d40d4f59f94_720w" src="https://user-images.githubusercontent.com/37235023/147567404-6248c0fc-a58b-4d83-87ef-78fcab1f01c7.png">
**定义1**：充分统计量（sufficient statistics）
考虑两个随机变量X、Y，如果存在一个映射关系S(X)使得它包含了所有X关于Y的信息，那么就称作S(X)是X关于Y的充分统计量，即I(S(X);Y)=I(X;Y)。
**定义2**：最小充分统计量（minimal sufficient statistics）
考虑两个随机变量X、Y，其最小充分统计量T(X)是所有重复呢统计量S(X)中包含关于X信息最少的统计量，即
<img width="745" alt="v2-5c37529d8a1e4e4865fe30d32ff04f38_720w" src="https://user-images.githubusercontent.com/37235023/147567686-32ffd169-fcd3-4ad6-acf0-0547582e1a01.png">

**Comment by Fassial** [comment]
## 深度学习中的信息论
可以把神经网络看作是一个信息处理的过程，每一层都对信息做了相应的处理。其中，Y->X是数据产生的过程，后面每一层都是神经网络的处理过程，最后是神经网络的输出。
![v2-b163f349b72a98ca0f6a80319e28d0f4_720w](https://user-images.githubusercontent.com/37235023/147567890-8c619483-6db3-45f7-a4c6-079f8958a968.jpg)
对于每一层T，都可以考虑两个量I(X;T)和I(T;Y)；而随着神经网络层数的不同，这些量满足如下不等式：
<img width="737" alt="v2-9d1d576f0913c9b9f9093dce711b455d_720w" src="https://user-images.githubusercontent.com/37235023/147567977-0e11dc99-139c-49c9-81a2-d4abdd0990c4.png">
考虑一个information plane，其横轴是I(X;T)、纵轴是I(T;Y)，那么神经网络中每一层都可以画成这个平面上的一个点，形成的样子大概是下图中绿线上的若干点：比如右上点表示T1，左下点表示Tk。

**Comment by Fassial** [comment]
如何定义深度学习模型中的“最优”表示？最优表示是对于输入X的某种处理，使得它关于预测的标签Y，是最小充分统计量。
最小充分统计量要求信息一点都不损失的情况下还所含信息量最小，要求比较严格，一般我们可以通过信息瓶颈（information bottleneck）来近似找到相应的最小充分统计量。也就是希望学习到encoder（p(t|x)）、decoder（p(y|t)）和相应表示层的先验分布p(t)满足如下最优化问题：
<img width="731" alt="v2-ff4b4db831f082c1b30edd1f5a8f401a_720w" src="https://user-images.githubusercontent.com/37235023/147620088-e197f7c6-2ee4-47f8-95c3-d7d58d13e2c3.png">
相应只有一个隐式解（注意到，它们之间还是搅合在一起的，没法分开），称该式子为IB self-consistent equations：
<img width="734" alt="v2-1d1582f934a3e2f5c1e22c5b4c55d7f2_720w" src="https://user-images.githubusercontent.com/37235023/147620256-b3d44bb9-93cc-4574-bde8-614a3e76562e.png">
其中Z是归一化函数。

**Comment by Fassial** [comment]
## 在信息平面上来看神经网络学习过程
下面文章做了一些实验，来在信息平面上观察神经网络的学习过程。
实验设定：
 - 数据：人造的二分类数据，方便互信息计算；正负样本均衡，I(X;Y)≈0.99。
 - 神经网络结构：7层MLP（12->10->7->5->4->3->2），没有外加其它正则。
 - 激活函数：最后一层sigmoid，其他层tanh。
 - 训练：SGD with cross-entropy loss。

**结论1**：不同层的表示满足公式5、6的不等式
左、中、右三图分别代表训练的不同过程，左下角橙色点代表比较靠后的神经网络层，右上角的绿色点代表比较靠前的神经网络层。
![v2-3c145bfa4e55a4973c7bd172a813485e_720w](https://user-images.githubusercontent.com/37235023/147620530-0119ac72-1494-4385-94ae-d7ec55a6703e.jpg)
**结论2**：神经网络的学习过程是先增大表示和标签之间的互信息（ERM阶段，empirical risk minimization），然后再减小表示和输入之间的互信息（表示压缩阶段，representation compression）。
**结论3**：当数据量比较小的时候，信息压缩阶段可能减小表示和标签之间的互信息（过拟合）。
左、中、右三图分别画了不同数据量下的神经网络训练过程。可以发现训练过程几乎可以分为两个阶段，两阶段之间的界限用绿色线分开。注意到，下图画出的点，都是在测试集上统计的。
![v2-542ca80f58c73c322580e1c6ab837722_720w](https://user-images.githubusercontent.com/37235023/147620728-b1180eb3-f585-4f07-ae32-469e576ddf2d.jpg)

**Comment by Fassial** [comment]
**结论4**：ERM阶段和表示压缩阶段的分界点可以由梯度信噪比刻画。
下图画出了训练过程中神经网络每一层梯度的均值和方差。在ERM阶段，神经网络梯度的信噪比较高，各层表示快速朝着最小化empirical risk的方向优化；而在表示压缩阶段，则方差开始变得更大。
![v2-490b3801061010a068c6ea5d593c0919_720w](https://user-images.githubusercontent.com/37235023/147620852-b2e379b5-d064-4315-8afc-acad5305b0ff.jpg)
**结论5**：神经网络层数更多的好处是能加速表示压缩阶段。
这里画出了不同神经网络层数下的学习过程，可以发现，层数越多，压缩阶段进行得越快。文章对此还是用random-diffusion模型进行了一些简单的分析。
![v2-b150e1c22f1a098fa4d3e89f90866e15_720w](https://user-images.githubusercontent.com/37235023/147620946-d8327447-2ec2-4ef1-9b45-4f7434b060f6.jpg)
**结论6**：学习收敛后，不同层表示对应self-consistent equation不同β下的解。
回忆一下，self-consistent equation对应的是信息瓶颈优化问题的最优解。即，在神经网络的训练下，各层表示能学习到相应的最优解，只不过对应的β数值不同；越靠后的表示层对应的β越小。
![v2-c65c7a565c555945dd70f2e623738f3c_720w](https://user-images.githubusercontent.com/37235023/147621116-71070d37-2565-4004-ab8a-3dbe220066c6.jpg)
**结论7**：训练数据量越小，最后学习到的表示在相对于标签的信息量越少。
![v2-5b6a6f0bbc65c69b98f7dd925485390f_720w](https://user-images.githubusercontent.com/37235023/147621167-e7e2b95f-6bac-4a47-aaf9-b4fec3982a04.jpg)

**Comment by Fassial** [comment]
## 不同的声音
在该篇文章中做了进一步实验，得出以下结论：
 - 表示压缩阶段的存在依赖于他们使用的tanh/sigmoid激活函数（双边饱和），但是如果换成线性或者单边饱和函数（比如ReLU），就没有了相应效果。
 - 表示压缩和泛化之间没有因果联系：没有表示压缩的神经网络也能泛化，压缩了也可能不泛化。真tm玄学，压缩之后可能把关键特征扔了，然后留下一堆noise？
 - 表示压缩阶段的出现和SGD没有必然联系，即使用full gradient训练也能出现表示压缩阶段。
 - 当输入包含任务相关信息以及任务无关信息时，随着训练的进行：
    1）表示层中的任务无关信息被压缩。
    2）表示层关于输入的信息缓慢增加。
    3）信息压缩发生在持续的整个训练过程，而不是一个单独的阶段。
